Autopilot/Src/PID.cpp

```diff
@@ -1,78 +1,72 @@
 /**
  * @file PID.cpp
- * @author Ian Frosst
- * @date  March 26, 2017
- * @copyright Waterloo Aerial Robotics Group 2017 \n
- *   https://raw.githubusercontent.com/UWARG/ZeroPilot-SW/devel/LICENSE.md
+ * @authors Ian Frosst, Anthony Berbari
  */
 
-#include "PID.h"
+#include "PID.hpp"
 
-/**
- * Filtering constant for derivative. Between 0 and 1.
- * The larger this is, the more twitchy D control is.
- */
-#define FILTER (0.4f)
+/***********************************************************************************************************************
+ * Code
+ **********************************************************************************************************************/
 
-/* Generic PID functions. Can be used to PID other things (flaps, etc) */
+PIDController::PIDController(float _kp, float _ki, float _kd, float _i_max, float _min_output, float _max_output){
+	kp = _kp;
+	kd = _kd;
+	ki = _ki;
 
+	i_max = _i_max;
```

Comment:
In PID.hpp i_max is declared with uint16_t, but the parameter "_i_max" is a float. This may result in truncation, which could be an issue. 

(if this isn't an issue, ignore please)

Reply:
good eye. Will do